test/test_bmp.cpp

```diff
@@ -0,0 +1,271 @@
+#include <catch2/catch_all.hpp>
+#include "../src/bmp.hpp"
+#include "../src/pixel.hpp"
+
+#include <bit>
+#include <iostream>
+
+using Catch::Matchers::ContainsSubstring;
+
```

Comment:
I'd also add some very basic tests, like creating a default `BMP` instance, or some select `BMP(width, height)` variant(s), and then checking e.g. `get_width`, `get_heigth` on them.  (That also automatically makes you consider edge cases, like "what is the behavior for a default-initialized `BMP dummy;`?")